### PR TITLE
chore: rustdoc + cleanup fuel-indexer-lib

### PR DESCRIPTION
**Pull request by ra0x3**
- [ ] Please add proper labels
- [ ] If there is an issue associated with this PR, please link the issue (right-hand sidebar)
- [ ] If there is not an issue associated with this PR, add this PR to the "Fuel Indexer" project (right-hand sidebar)

#### ⚠️ I'm going through all crates and fixing hacks, cleaning up, adding clarity and rustdoc comments. Starting with `fuel-indexer-lib`

- For merge after https://github.com/FuelLabs/fuel-indexer/pull/999
- This cleans up the fuel-indexer-lib module
- Adds rustdoc comments
- Updates README briefly
- **No code logic is changed, code is only moved around for clarity**

### Testing steps

- N/A

### Changelog

- chore: rustdoc + cleanup fuel-indexer-lib

### CODE REVIEW

File: packages/fuel-indexer-lib/src/config/cli.rs [hunk 1]
```diff
@@ -0,0 +1,300 @@
+pub use crate::{
```
Comment (ra0x3) [inline packages/fuel-indexer-lib/src/config/cli.rs:1]: This module is configuration related to the CLI.

File: packages/fuel-indexer-lib/src/config/mod.rs [hunk 2]
```diff
@@ -66,292 +45,29 @@ pub fn env_or_default(var: EnvVar, default: String) -> String {
     std::env::var(var.as_ref()).unwrap_or(default)
 }
 
-#[derive(Debug, Parser, Clone)]
```
Comment (ra0x3) [inline packages/fuel-indexer-lib/src/config/mod.rs:69]: Removed the cli-related stuff to its own module

File: packages/fuel-indexer-lib/src/utils.rs [hunk 3]
```diff
@@ -229,71 +236,3 @@ pub async fn init_logging(config: &IndexerConfig) -> anyhow::Result<()> {
     }
     Ok(())
 }
-
-pub fn format_exec_msg(exec_name: &str, path: Option<String>) -> String {
```
Comment (ra0x3) [inline packages/fuel-indexer-lib/src/utils.rs:233]: These functions are only used by forc-index so they were moved to forc-index

File: packages/fuel-indexer-api-server/src/uses.rs [hunk 4]
```diff
@@ -86,7 +86,7 @@ pub(crate) async fn get_fuel_status(config: &IndexerConfig) -> ServiceStatus {
                 .await
                 .unwrap_or_default();
 
-            let fuel_node_health: FuelNodeHealthResponse =
+            let fuel_node_health: FuelClientHealthResponse =
```
Comment (deekerno) [inline packages/fuel-indexer-api-server/src/uses.rs:89]: I see that we're referring to `FuelClient*`, which is clearer; can we also rename `fuel_node_health` to `fuel_client_health` to match?

File: packages/fuel-indexer-tests/src/fixtures.rs [hunk 5]
```diff
@@ -287,7 +287,7 @@ pub async fn api_server_app_postgres(
         verbose: true,
         local_fuel_node: false,
         indexer_net_config: false,
-        fuel_node: FuelNodeConfig::default(),
+        fuel_node: FuelClientConfig::default(),
```
Comment (deekerno) [inline packages/fuel-indexer-tests/src/fixtures.rs:290]: Can we also rename `fuel_node` to `fuel_client`?

File: packages/fuel-indexer-api-server/src/uses.rs [hunk 6]
```diff
@@ -22,8 +22,8 @@ use fuel_indexer_lib::{
     config::{auth::AuthenticationStrategy, IndexerConfig},
     defaults,
     utils::{
-        AssetReloadRequest, FuelNodeHealthResponse, IndexStopRequest, ServiceRequest,
-        ServiceStatus,
+        FuelClientHealthResponse, ReloadRequest, ServiceRequest, ServiceStatus,
```
[the remark this answers is not in the document]
Comment (deekerno) [reply packages/fuel-indexer-api-server/src/uses.rs:25]: Agreed.

File: packages/fuel-indexer-lib/src/utils.rs [hunk 7]
```diff
@@ -67,36 +66,40 @@ pub fn local_repository_root() -> Option<String> {
...
 #[derive(Debug)]
-pub struct AssetReloadRequest {
+pub struct ReloadRequest {
     pub namespace: String,
     pub identifier: String,
 }
 
+/// Request to remove the specified indexer executor from the indexer service.
+///
+/// Sent from API server to indexer service.
```
Comment (deekerno) [inline packages/fuel-indexer-lib/src/utils.rs:81]: We can probably remove this line as it's already mentioned in the doc comment for `ServiceRequest`.

File: packages/fuel-indexer-lib/src/utils.rs (hunk 3 above)
Comment (deekerno) [reply packages/fuel-indexer-lib/src/utils.rs:233]: Nice. 👌🏽

**Review by deekerno — CHANGES_REQUESTED** [review]
Looks great! Just left some comments about consistency and redundancy.

File: packages/fuel-indexer-api-server/src/uses.rs (hunk 4 above)
Comment (ra0x3) [reply packages/fuel-indexer-api-server/src/uses.rs:89]: @deekerno  That's the thing. If we rename _everything_ to `FuelClient` the footprint is obviously much larger, and includes a few breaking changes. I originally actually had this (changing _everything_) but felt that was a little too large of a footprint for this cleanup - and that should be in its own separate breaking PR.

@lostman If we're all aligned to change these namings to `FuelClient*` let's do that separately

File: packages/fuel-indexer-tests/src/fixtures.rs (hunk 5 above)
Comment (ra0x3) [reply packages/fuel-indexer-tests/src/fixtures.rs:290]: [See comment](https://github.com/FuelLabs/fuel-indexer/pull/1007#discussion_r1238597585)

**Review by deekerno — APPROVED** [review]
Approving in light of @ra0x3's comment.

**Review by deekerno — APPROVED** [review]

**Pull request merged**